PapyrusCore/Sources/Request.swift

```diff
@@ -6,3 +6,44 @@ public protocol Request {
     var headers: [String: String] { get set }
     var body: Data? { get set }
 }
+
+
+public extension Request {
+    /// Create a cURL command from this instance
+    ///
+    /// - Parameter sortedHeaders: sort headers in output
+    /// - Returns: cURL Command
+    func curl(sortedHeaders: Bool = false) -> String {
+        let lineSeparator = " \\\n"
+        var components = [String]()
+
+        // Add URL on same line
+        if let url = self.url {
+            components.append("curl '\(url.absoluteString)'")
+        } else {
+            components.append("curl")
+        }
+
+        // Add method
+        components.append("-X \(method)")
+
+        // Add headers
+        if sortedHeaders {
+            for (key, value) in headers.sorted(by: { $0.key < $1.key }) {
+                components.append("-H '\(key): \(value)'")
+            }
+        } else {
+            for (key, value) in headers {
+                components.append("-H '\(key): \(value)'")
+            }
+        }
```

Comment:
Slightly cleaner option that I believe works:

```suggestion
        let headerOptions = headers.map { "-H '\($0): \($1)'" }
        components += sortedHeaders ? headerOptions.sorted() : headerOptions
```

Reply:
very nice. this has the added benefit of also sorting by value two headers have the same key